en/6-persistence/README2.md

```diff
@@ -0,0 +1,505 @@
+The current implementation of NEO in C# uses LevelDB, a fast key-value database, to persist blockchain information. This DB is used for both system data, like blocks and transactions, but also for smart contract data.  
+When the node receives a message that triggers a change in state (like a new block or header), it retrieves and updates a snapshot, commiting its results when the operation is done.   
+Below we have a simplified example of how this is happens:
+
+![snapshot](persistence_snapshot.png)
+
+
+
+
+#### LevelDB Table Structure
+The data is stored using the structured provided by the classes from the [Ledger](https://github.com/neo-project/neo/tree/master-2.x/neo/Ledger) package. Using a `prefix` to separate each 'table'.
+
+We can see here the prefixes used in neo (C#):
+
+
+| Prefix     | Description     | LevelDB Key | Level DB Value
+| :------------- | :------------- | :------------- | :------------- |
+| 0x01      | **Block**: Block header prepended with `system fee`     | Block Hash | [BlockState](https://github.com/neo-project/neo/blob/master-2.x/neo/Ledger/BlockState.cs) |
+| 0x02      | **Transaction**: Transaction prepended with `block index`       | Transaction Hash | [TransactonState](https://github.com/neo-project/neo/blob/master-2.x/neo/Ledger/TransactionState.cs) |
+| 0x40      | **Account**: account meta-data (votes, balances)       | Account Script Hash | [AccountState](https://github.com/neo-project/neo/blob/master-2.x/neo/Ledger/AccountState.cs) |
+| 0x44      | **Unspent Coins**: Transactions that can be used as input in another transaction       | Transaction Hash | [UnspentCoinState](https://github.com/neo-project/neo/blob/master-2.x/neo/Ledger/UnspentCoinState.cs) |
+| 0x45      | **Spent Coins**: Transactions used as input in a previous transaction (spent)       | Transaction Hash | [SpentCoinState](https://github.com/neo-project/neo/blob/master-2.x/neo/Ledger/SpentCoinState.cs) |
+| 0x48      | **Validators**: Network Validators meta-data        | Validator Public Key | [ValidatorState](https://github.com/neo-project/neo/blob/master-2.x/neo/Ledger/ValidatorState.cs)|
+| 0x4c      | **Assets**: Native coins meta-data       | Register Transaction Hash | [AssetState](https://github.com/neo-project/neo/blob/master-2.x/neo/Ledger/AssetState.cs) |
+| 0x50      | **Contracts**: Contract meta-data       | Contract Script Hash| [ContractState](https://github.com/neo-project/neo/blob/master-2.x/neo/Ledger/ContractState.cs) |
+| 0x70    | **Smart Contract Storage**: Storage used by SmartContracts       | ScriptHash + Key | Byte[] |
+| 0x80      | **Pending Headers**: Used in synchronization to track blocks waiting to be synced        | Prefix only | [HeaderHashList](https://github.com/neo-project/neo/blob/master-2.x/neo/Ledger/HeaderHashList.cs) |
+| 0x90      | **Validators Count**: TODO         | Prefix only| [ValidatorsCountState](https://github.com/neo-project/neo/blob/master-2.x/neo/Ledger/ValidatorsCountState.cs) |
```

Comment:
`Validators Count` is used for recording the votes of  how many consensus nodes to be selected.  More detail about voting: https://docs.neo.org/developerguide/en/articles/consensus/vote_validator.html